pkg/api/staking_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/http"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/staking/stakingcontract"
+	stakingContractMock "github.com/ethersphere/bee/pkg/staking/stakingcontract/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func TestDepositStake(t *testing.T) {
+	t.Parallel()
+
+	addr := swarm.MustParseHexAddress("f30c0aa7e9e2a0ef4c9b1b750ebfeaeb7c7c24da700bb089da19a46e3677824b")
+
+	minStake := big.NewInt(1).String()
+	minStakedAmount := stakingcontract.MinimumStakeAmount
+	depositStake := func(address string, amount string) string {
+		return fmt.Sprintf("/stake/deposit/%s/%s", address, amount)
+	}
+
+	t.Run("ok", func(t *testing.T) {
+		t.Parallel()
+
+		contract := stakingContractMock.New(
+			stakingContractMock.WithDepositStake(func(ctx context.Context, stakedAmount big.Int, overlay swarm.Address) error {
+				if stakedAmount.Cmp(minStakedAmount) == -1 {
+					return stakingcontract.ErrInsufficentStakeAmount
+				}
+				return nil
+			}),
+		)
+		ts, _, _, _ := newTestServer(t, testServerOptions{DebugAPI: true, StakingContract: contract})
+		jsonhttptest.Request(t, ts, http.MethodPost, depositStake(addr.String(), minStake), http.StatusOK)
+	})
+
+	t.Run("with invalid stake amount", func(t *testing.T) {
+		t.Parallel()
+
+		invalidMinStake := big.NewInt(0).String()
+		invalidMinStakedAmount := big.NewInt(1)
+		contract := stakingContractMock.New(
+			stakingContractMock.WithDepositStake(func(ctx context.Context, stakedAmount big.Int, overlay swarm.Address) error {
+				if stakedAmount.Cmp(invalidMinStakedAmount) == -1 {
+					return stakingcontract.ErrInsufficentStakeAmount
+				}
```

Comment:
consider simplifying this mock function by always returning `ErrInsufficentStakeAmount`